src/Digdir.Domain.Dialogporten.Application/Features/V1/ServiceOwner/Dialogs/Commands/Create/CreateDialogCommandValidator.cs

```diff
@@ -171,8 +173,16 @@ public CreateDialogDialogElementUrlDtoValidator()
             .MaximumLength(Constants.DefaultMaxUriLength);
         RuleFor(x => x.MimeType)
             .MaximumLength(Constants.DefaultMaxStringLength);
+        RuleFor(x => x.MimeType)
+            .Equal(MediaTypeNames.Text.Markdown)
+            .When(x => x.FrontChannelEmbed)
+            .WithMessage($"'{{PropertyName}}' must be '{MediaTypeNames.Text.Markdown}' when {nameof(DialogElementUrl.FrontChannelEmbed)} is set to true.");
         RuleFor(x => x.ConsumerType)
             .IsInEnum();
+        RuleFor(x => x.ConsumerType)
+            .Equal(DialogElementUrlConsumerType.Values.Gui)
+            .When(x => x.FrontChannelEmbed)
+            .WithMessage($"'{{PropertyName}}' must be {DialogElementUrlConsumerType.Values.Gui} when {nameof(DialogElementUrl.FrontChannelEmbed)} is set to true.");
```

Comment:
@elsand Had a chat with @MagnusSandgren here
Is the FrontChannelEmbed flag really needed, or can frontend just assume that if an element has mime markdown it is meant to be embedded? 

Or, is there a scenario where ServiceOwner would use Markdown mime and NOT want it embedded .. it would basically download markdown as a file?

Reply:
We need to consider that we at we at some point would like to support other types of content besides Markdown. Most likely this will be some custom format that should be rendered in a particular way, eg. payment information or a simple form, but I suppose also videos or images might be in some cases natural to embed (and other cases not). A way to do this explicitly without a separate field might be to use a vendor-prefixed MIME-type with its own "type" parameter, as allowed in RFC6838? Ie

```
application/vnd.dialogporten.frontchannelembed+json;type=markdown
application/vnd.dialogporten.frontchannelembed+json;type=payment
application/vnd.dialogporten.frontchannelembed+json;type=simpleform
application/vnd.dialogporten.frontchannelembed+json;type=restrictedhtml
application/vnd.dialogporten.frontchannelembed+json;type=video
```
Thoughts?

Reply:
Sounds good, then we can leave it with no flag, but still have explicit embed types.
Have a list of allowed types in the validators, and then let frontend embed based on these?

Reply:
@elsand 
Also, `Sørge for at dialogToken blir generert hvis frontChannelEmbed oppgis`, from the issue,
does that mean dialogToken *only* should be set when there are any URLs of this type? 
(Or a write action, for GUI actions)

Reply:
> Sounds good, then we can leave it with no flag, but still have explicit embed types?
> Have a list of allowed types in the validators, and then let frontend embed based on these?

Right, we should allow any MIME type per se, but it probably makes sense to validate those frontchannelembed-types (which should be synced with what Arbeidsflate supports, which for now is just the markdown variant)

> Also, Sørge for at dialogToken blir generert hvis frontChannelEmbed oppgis, from the issue,
does that mean dialogToken only should be set when there are any URLs of this type?
(Or a write action, for GUI actions)

No, disregard that. Generating a token is cheap enough so we do it every time, and it might be used for any action/dialogelement.

Reply:
Hm, I dunno some of these, [f.ex. under "Font"](https://github.com/dotnet/runtime/blob/2a80808f86459a95229c6f9c85ce32b62636fceb/src/libraries/System.Net.Mail/src/System/Net/Mime/MediaTypeNames.cs#L61), doesn't make much sense? 
Committed new suggestion

Reply:
Not sure if it makes sense to support "text/javascript" either, but this works fine for now. We can always change the list of valid  media types later.